spec/services/event_log/move_runner_spec.rb

```diff
@@ -5,55 +5,87 @@
 RSpec.describe EventLog::MoveRunner do
   subject(:runner) { described_class.new(move) }
 
-  let!(:move) { create(:move, to_location: original_location) }
-  let!(:event1) do
-    create(
-      :move_event,
-      event_type,
-      eventable: move,
-      client_timestamp: event1_timestamp,
-      created_at: 100.minutes.ago,
-      details: { note: 'Event1', event_params: { relationships: { to_location: { data: { id: event1_location.id } } } } },
-    )
+  let!(:move) { create(:move, to_location: original_location, status: move_status) }
+  let(:original_location) { create(:location) }
+  let(:move_status) { 'requested' }
+
+  before { allow(Notifier).to receive(:prepare_notifications) }
+
+  shared_examples_for 'it calls the Notifier with an update_status action_name' do
+    before { runner.call }
+
+    it 'calls the Notifier with update_status' do
+      expect(Notifier).to have_received(:prepare_notifications).with(topic: move, action_name: 'update_status')
+    end
   end
-  let!(:event2) do
-    create(
-      :move_event,
-      event_type,
-      eventable: move,
-      client_timestamp: event2_timestamp,
-      created_at: 1.minute.ago,
-      details: { note: 'Event2', event_params: { relationships: { to_location: { data: { id: event2_location.id } } } } },
-    )
+
+  shared_examples_for 'it calls the Notifier with an update action_name' do
+    before { runner.call }
+
+    it 'calls the Notifier with update' do
+      expect(Notifier).to have_received(:prepare_notifications).with(topic: move, action_name: 'update')
+    end
   end
 
-  let(:event_type) { :redirect }
-  let(:original_location) { create(:location) }
-  let(:event1_location) { create(:location, title: 'Event1-Location') }
-  let(:event2_location) { create(:location, title: 'Event2-Location') }
-  let(:event1_timestamp) { '2020-05-22 09:00:00' }
-  let(:event2_timestamp) { '2020-05-22 10:00:00' }
-
-  context 'when events are received in a chronological order' do
-    it 'updates the move to event2 redirect location' do
-      expect { runner.call }.to change(move, :to_location).from(original_location).to(event2_location)
+  shared_examples_for 'it does not call the Notifier' do
+    before { runner.call }
+
+    it 'calls the Notifier with update' do
+      expect(Notifier).not_to have_received(:prepare_notifications)
     end
   end
 
-  context 'when events are not received in a chronological order' do
-    let(:event1_timestamp) { '2020-05-22 10:00:00' } # chronological second event received first
-    let(:event2_timestamp) { '2020-05-22 09:00:00' } # chronological first event received last
+  context 'when event_name=redirect' do
+    let!(:event1) { create(:move_event, :redirect, eventable: move, client_timestamp: event1_timestamp, created_at: 100.minutes.ago, details: event1_details) }
+    let(:event1_details) { { note: 'Event1', event_params: { relationships: { to_location: { data: { id: event1_location.id } } } } } }
```

Comment:
These are a bit tricky to read, maybe break over multiple lines? Same on line 44.

Reply:
👍